### ISSUE TITLE
论文中性能是否是添加了DSL后处理的结果

### ISSUE THREAD

**Issue by M4RK-Young** [musicman217/Text-Proxy#6]
作者您好，我在复现代码的时候发现有两个性能，一个是添加了DSL后处理的性能，一个是没添加DSL后处理的性能，但是论文的表格中也没有表明，所以想请问您论文中采用的是哪个性能

**Comment by musicman217** [comment]
你好，感谢你的关注
论文中使用的是没有添加DSL后处理的结果

**Comment by M4RK-Young** [comment]
您好 我在msrvtt数据集上用clip32的backbone测出来最好的性能是50.7 离文章里52.3的性能还是有点远 我已经尽可能按照readme文件配置环境里 请问还可能是什么原因呢

![Image](https://github.com/user-attachments/assets/75333e76-eb7b-4738-9a3a-f3e02c4049d7)

**Comment by M4RK-Young** [comment]
我遇到了和另一个open问题一样的bug 出现了除0错误 逐行debug之后发现是一个clip_vip的一个线性层没有初始化self.visual_projection_patch = nn.Linear(self.vision_embed_dim, self.projection_dim, bias=False)也就是这行代码 然后我添加了对它初始化的代码
nn.init.normal_(
                    module.visual_projection_patch.weight,
                    std=module.vision_embed_dim**-0.5 * self.config.initializer_factor,
                ) 就这个地方进行了改动 不知道是不是这个原因

**Comment by M4RK-Young** [comment]
我现在的设备是双卡l40

**Comment by M4RK-Young** [comment]
我好像解决问题了 我又试了一下不提前压缩视频 最后测得51.9

**Comment by M4RK-Young** [comment]
感谢您的回复，我是用apex跑的，我现在又试了一下clip16的backbone，压不压缩视频都试了，结果最好都只有53，离文中的55.2还有距离，我只修改了backnone别的配置都没做修改，不知道是什么原因，是否还要修改其它内容。

**Comment by musicman217** [comment]
你好，patch16主要的区别在于direction leader的两个系数都设置为-1了，我今晚会上传一下patch16的conf，如果方便你可以手动设置下两个系数，这里论文的最后一个消融也有说明可以参考一下

**Comment by M4RK-Young** [comment]
好的 非常感谢您的回复！

**Comment by musicman217** [comment]
你好，我明天会把conf和text_proxy.py文件重新上传一下，这两天有点忙，你可以改一下text_proxy.py的173行附近的 dir_vec = text_feat.unsqueeze(1) - pro_proxies，改为dir_vec = -1 * text_feat.unsqueeze(1) - (-1) * pro_proxies，注意dash_option的部分为theta不要修改，具体可以参考论文，论文里陈述了全部训练细节
---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2025年07月01日 09:45 |
| 收件人 | musicman217/Text-Proxy ***@***.***> |
| 抄送人 | Jian ***@***.***>,
Comment ***@***.***> |
| 主题 | Re: [musicman217/Text-Proxy] 论文中性能是否是添加了DSL后处理的结果 (Issue #6) |
M4RK-Young left a comment (musicman217/Text-Proxy#6)

我观察到代码里那两个参数并不是显式设置的，所以我按照自己的理解进行了修改
if self.cfg.dash_option == "linear":
dash = self.dash_weights(frame_logits) # (a,b,f)->(a,b,512) or (a=b,1,f)->(a=b,1,512)
dash = torch.exp(dash)
elif self.cfg.dash_option == "theta":
dash = torch.exp(self.dash_factor * frame_logits.mean(2)).unsqueeze(2) # -> (a,b,1)
elif self.cfg.dash_option == "fixed": # 修改部分 手动设置两个超参数为-1
# Use custom delta and eta values: d = δtq - ηdlk
delta = self.cfg.get("delta", -1.0) # default -1
eta = self.cfg.get("eta", -1.0) # default -1

        # Recalculate dir_vec with custom delta and eta
        dir_vec = delta * text_feat.unsqueeze(1) - eta * pro_proxies
        dir_vec = dir_vec / torch.norm(dir_vec, p=2, dim=2, keepdim=True)

        # Use fixed dash value or similarity-based dash
        if hasattr(self.cfg, 'fixed_dash_value'):
            dash = torch.full((dir_vec.shape[0], dir_vec.shape[1], 1),
                            self.cfg.fixed_dash_value, device=dir_vec.device)
        else:
            dash = torch.exp(self.dash_factor * frame_logits.mean(2)).unsqueeze(2)

    text_proxy = text_feat.unsqueeze(1) + dash * dir_vec    


我不知道我这样修改是否正确，还是想请您提供一下patch16的config，以及这两个超参数修改的方法

**Comment by M4RK-Young** [comment]
您好 我按照您给出的配置文件和text_proxy.py文件测完只得到了53.2的性能 不知道是在什么原因了 还有一个问题是这两个参数改成-1之后训练时间成倍上升 从不到5小时上升到了30几个小时 不知道这是否正常

**Comment by musicman217** [comment]
> 您好 我按照您给出的配置文件和text_proxy.py文件测完只得到了53.2的性能 不知道是在什么原因了 还有一个问题是这两个参数改成-1之后训练时间成倍上升 从不到5小时上升到了30几个小时 不知道这是否正常

改为-1不可能会发生训练时间变长的情况，请排查是否是batchsize设置过大或者卡数过多的缘故，我给出的patch16的conf中设置为batchsize为56，注意conf里的batchsize是指单卡batchsize，不是所有卡的和，同时论文中写明msrvtt的patch16实验是在一张A800上以batchsize=56的情况进行的实验，最后注意text_proxy.py文件的修改

**Comment by M4RK-Young** [comment]
好的 感谢您的回复

**Comment by M4RK-Young** [comment]
可能是我之前batchsize设置小了 我最后用A100复现出了55.1的性能！
